### PR TITLE
improve ByteUtil printable char validation of multi-byte charset

### PR DESCRIPTION
**Pull request by fbruton**
~~Planning to add additional unit-tests.~~

### CODE REVIEW

File: src/test/java/emissary/util/ByteUtilTest.java [hunk 1]
```diff
@@ -208,4 +209,26 @@ void testGrabLine() {
         assertEquals("This is line three", ByteUtil.grabLine(data, 35), "Last line extraction");
     }
 
+    @Test
+    void testHasnonPrintableValues() {
+        String newLineCarriageTab = "\tThis is line one\r\nThis is line two\nThis is line three\n\nEnding with a tab";
+        assertFalse(ByteUtil.hasNonPrintableValues(newLineCarriageTab.getBytes(StandardCharsets.UTF_8)));
```
Comment (ldhardy) [inline src/test/java/emissary/util/ByteUtilTest.java:215]: It's possible there is more than one definition, but I thought non-printing characters were tab, newline, etc.  Maybe I need more concise definition of what we're trying to identify.

Comment (ldhardy) [reply src/test/java/emissary/util/ByteUtilTest.java:215]: Also, is it supposed to end with a tab?  \t?

Comment (fbruton) [reply src/test/java/emissary/util/ByteUtilTest.java:215]: Maybe it should be called binary.

Comment (fbruton) [reply src/test/java/emissary/util/ByteUtilTest.java:215]: The previous code allowed tabs, new lines and carriage returns, which makes sense.

File: src/main/java/emissary/util/ByteUtil.java [hunk 2]
```diff
@@ -290,20 +290,104 @@ public static String grabLine(byte[] data, int pos) {
...
      * Scans a byte array looking for non-printable values.
      * 
-     * @param bytes the bytes to be scanned.
-     * @return whether or not there were non-printable values.
+     * @param utf8Bytes the bytes to be scanned.
+     * @return true if the byte array contains an invalid text character.
      */
-    public static boolean hasNonPrintableValues(final byte[] bytes) {
-        boolean badCharacters = false;
+    public static boolean hasNonPrintableValues(final byte[] utf8Bytes) {
```
Comment (ldhardy) [inline src/main/java/emissary/util/ByteUtil.java:296]: Per our discussion in the office, Dave and I would like a chance to re-work this entire method and move it elsewhere.  We believe it should live in the test jar, maybe as a static member of RegressionTest or some other test utility.  I confirmed it's not used upstream in any production code.  But other members of this class are used in production code, so we should leave the class and it's other methods here.

**Review by ldhardy — APPROVED** [review]

**Comment by dev-mlb** [comment]
I'm thinking we should merge the ideas from this pr and #1006

**Review by cfkoehler — APPROVED** [review]
My experience in this area is pretty minimal. But the code makes sense to me and seems to do what is needed.

**Pull request closed without merging**